qiskit/compiler/transpiler.py

```diff
@@ -240,6 +240,9 @@ def callback_func(**kwargs):
     arg_circuits_list = isinstance(circuits, list)
     circuits = circuits if arg_circuits_list else [circuits]
 
+    if not circuits:
```

Comment:
Are you concerned at all about the side effects of `transpile(0), `transpile(False)` or `transpile(None)` here? I only ask because if a user did any of those I'd expect an error.

Reply:
Right before these lines, `circuits` is promoted to a list if it isn't one already, so in those cases, it'd all be promoted to `circuits = [0]`, `circuits = [False]`, etc.  I didn't check to see if super invalid output fails somewhere else, but the boolean coercion shouldn't cause any more problems here.

Reply:
Ok cool I see that now on L241. Heh I should have looked at more than just strict diff to see the surrounding context.